sysl2/sysl/integration/deps.go

```diff
@@ -0,0 +1,284 @@
+package integration
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/anz-bank/sysl/src/proto"
+	"github.com/anz-bank/sysl/sysl2/sysl/utils"
+	"github.com/yale8848/stream"
+)
+
+type AppElement struct {
+	Name     string
+	Endpoint string
+}
+
+type AppDependency struct {
+	Self   *AppElement
+	Target *AppElement
+}
+
+type DependencySet struct {
```

Comment:
@JackLeeHal Do you really need the struct `DependencySet`?
And this is a list now, not a set.

Reply:
It's a set actually, it will call Contain() before it call Add().